### DISCUSSION TITLE
Is there an easier way to validate configuration before running Renovate?

### DISCUSSION THREAD

**Discussion by kimsey0** [renovatebot/renovate#27145] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.176

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### What is your question?

Our company has a number of Engineering teams, each with a couple of Azure Repos they manage themselves. We have a central Renovate pipeline in Azure Pipelines which spins up a job per repository and runs Renovate through Docker for that one repository. We would like the job to fail if the team has checked in an invalid `renovate.json` in their repository.

Optimally, we could configure Renovate to log an error and exit with a non-zero exit code in this case, but there is no such setting. An invalid `renovate.json` file is only logged as a info-level message and doesn't affect the exit code, so we'd have to scan through all the log lines to find out if the file is invalid.

Instead, we plan to run `renovate-config-validator`, also through the Docker image. Here, the validator won't check out the repository and find the `renovate.json` file, so it looks like we have to manually check out the repository outside of Docker and then mount it as a volume for `renovate-config-validator` to validate. That also means that, in the vast majority of cases where the configuration is valid, we end up checking the repository out twice, once for `renovate-config-validator`, and then once again in the actual Renovate run.

Is there an easier way for us to achieve what we want?

**Comment by RahulGautamSingh** [comment]
We treat repository configuration errors as warnings and do not exit on those. It is a design choice. If you want to change that then you need to change this logger from info to error.
https://github.com/renovatebot/renovate/blob/d424c7f1305af6b1c476099f9b5dc99b4edb2c05/lib/config/migrate-validate.ts#L40-L42

**Reply by RahulGautamSingh** [reply, marked as answer]
You could also wait for this feature: https://github.com/renovatebot/renovate/issues/26723

**Reply by kimsey0** [reply]
I don't think we want to fork Renovate just to change this log level. Is there an easier way to validate the configuration using the existing Docker images without having to manually check out the repository first?

**Reply by kimsey0** [reply]
Thanks for that issue link. I hadn't seen it until after I wrote the comment above. I have subscribed to it.
